tools/build_script_generator/module.md

```diff
@@ -120,6 +120,70 @@ Program received signal SIGINT, Interrupt.
+
+Black Magic Probe creates two serial devices, the first being the GDB interface and the second a plain serial adaptor for debugging purposes.
+
+```
+$ ls -l /dev/tty.usb*
+crw-rw-rw-  1 root  wheel   21, 104 Feb 19 09:46 /dev/tty.usbmodemD6CE8AB1
+crw-rw-rw-  1 root  wheel   21, 106 Feb 19 09:46 /dev/tty.usbmodemD6CE8AB3
+```
+
+You can upload your code using a Black Magic Probe with specifying the GDB interface as `port` parameter.
+
+```
+$ scons bmp port=/dev/cu.usbmodemD6CE8AB1 verbose=1
+scons: Reading SConscript files ...
+bmp port iss = /dev/cu.usbmodemD6CE8AB1
+scons: done reading SConscript files.
+scons: Building targets ...
+arm-none-eabi-gdb -ex "target extended-remote /dev/cu.usbmodemD6CE8AB1" -ex "monitor swdp_scan" -ex "attach 1" -ex "load /Users/usr/Dev/Robotise/can-sync/build/can/release/src.elf" -ex "detach" -ex "quit"
+GNU gdb (GNU Tools for Arm Embedded Processors 8-2018-q4-major) 8.2.50.20181213-git
+Copyright (C) 2018 Free Software Foundation, Inc.
+License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
+This is free software: you are free to change and redistribute it.
+There is NO WARRANTY, to the extent permitted by law.
+Type "show copying" and "show warranty" for details.
+This GDB was configured as "--host=x86_64-apple-darwin10 --target=arm-none-eabi".
+Type "show configuration" for configuration details.
+For bug reporting instructions, please see:
+<http://www.gnu.org/software/gdb/bugs/>.
+Find the GDB manual and other documentation resources online at:
+    <http://www.gnu.org/software/gdb/documentation/>.
+
+For help, type "help".
+Type "apropos word" to search for commands related to "word".
+Remote debugging using /dev/cu.usbmodemD6CE8AB1
+Target voltage: unknown
+Available Targets:
+No. Att Driver
+ 1      STM32F1 medium density
+Attaching to Remote target
+warning: No executable has been specified and target does not support
+determining executable automatically.  Try using the "file" command.
+0x0800038e in ?? ()
+Loading section .vector_rom, size 0xec lma 0x8000000
+Loading section .text, size 0x10c0 lma 0x80000f0
+Loading section .rodata, size 0x15c lma 0x80011b0
+Loading section .hardware_init, size 0x8 lma 0x800130c
+Loading section .fastcode, size 0x58 lma 0x8001314
+Loading section .data, size 0x6c lma 0x800136c
+Loading section .fastdata, size 0xc lma 0x80013d8
+Loading section .table.zero.intern, size 0x8 lma 0x80013e4
+Loading section .table.copy.intern, size 0xc lma 0x80013ec
+Loading section .table.section_heap, size 0xc lma 0x80013f8
+Start address 0x8000e6c, load size 5120
+Transfer rate: 10 KB/sec, 365 bytes/write.
+Detaching from program: , Remote target
+[Inferior 1 (Remote target) detached]
+scons: done building targets.
+
+```
```

Comment:
Please reduce the verbosity of this output and remove personal information like paths.
I did something very similiar for the OpenOCD output here: https://github.com/modm-io/modm/blob/develop/tools/build_script_generator/module.md#using-openocd